server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -267,7 +267,7 @@ public Sink getSink(long timestamp)
                           throw new ISE("No timeline entry at all!");
                         }
 
-                        final Sink theSink = holder.getObject().getChunk(0).getObject();
+                        final Sink theSink = holder.getObject().iterator().next().getObject();
```

Comment:
The realtime plumber always uses SingleElementPartitionChunk, so there actually is no bug here to fix.